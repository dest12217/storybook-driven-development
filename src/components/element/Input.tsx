import * as React from 'react';

export default function Input(props: React.ComponentProps<'input'>) {
  return <>
    <style href="element-input" precedence="medium">{`
      .element-input {
        display: block;
        box-sizing: border-box;
        width: 100%;
        border-radius: 4px;
        border: 1px solid white;
        padding: 10px 15px;
        margin-bottom: 10px;
        font-size: 14px;
        background-color: #fff;
      }
    `}</style>
    <input {...props} className="element-input" />
  </>;
}
