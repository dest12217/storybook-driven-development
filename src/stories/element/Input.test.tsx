import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import type { ComposedStoryFn } from 'storybook/internal/types';
import { describe, expect, test } from 'vitest';
import * as stories from '@/stories/element/Input.stories';

const composedStories = composeStories(stories);

describe('Input', () => {
  describe('Snapshot', () => {
    test.each(Object.entries<ComposedStoryFn>(composedStories))('Snapshot: %s', (name, Story) => {
      const { asFragment } = render(<Story />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
