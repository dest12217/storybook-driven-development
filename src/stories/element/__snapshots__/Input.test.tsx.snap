// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > Snapshot > Snapshot: Default 1`] = `
<DocumentFragment>
  <input
    class="element-input"
  />
</DocumentFragment>
`;
